src/riak_kv_qry_worker.erl

```diff
@@ -168,6 +170,13 @@ make_key_conversion_fun(Table) ->
             Key
     end.
 
+%% Convert the sql select record to a version that is safe to pass around the
+%% cluster. Do not treat the result as a record in the local node, just as a
+%% tuple.
+convert_query_to_cluster_version(Query) ->
+    Version = riak_core_capability:get({riak_kv, sql_select_version}),
```

Comment:
@hmmr That would just get the latest version of the select record, not the agreed upon current version in the cluster.